File: frontend/src/api/studyPlan.ts
import { apiClient } from './client';
import type { ResourceItem, StudyPlanResponse } from '../types/studyPlan';

export async function generateStudyPlan(courseId: string, weeksAvailable = 15) {
  const { data } = await apiClient.post<StudyPlanResponse>('/api/study-plan/generate', {
    course_id: courseId,
    weeks_available: weeksAvailable,
  });

  return data;
}

export async function getStudyPlan(courseId: string) {
  const { data } = await apiClient.get<StudyPlanResponse>(`/api/study-plan/${courseId}`);
  return data;
}

export async function generateResources(courseId: string, topic?: string) {
  const { data } = await apiClient.post<ResourceItem[]>('/api/resources/generate', {
    course_id: courseId,
    topic,
  });

  return data;
}

export async function getResources(courseId: string) {
  const { data } = await apiClient.get<ResourceItem[]>(`/api/resources/${courseId}`);
  return data;
}

File: backend/app/routers/study_plan.py
from fastapi import APIRouter, Depends
from typing import Optional
from app.models.study_plan import GenerateStudyPlanRequest, StudyPlanResponse
from app.core.dependencies import get_guest_id

router = APIRouter()


@router.post("/generate", response_model=StudyPlanResponse)
async def generate_study_plan(
    body: GenerateStudyPlanRequest,
    guest_id: Optional[str] = Depends(get_guest_id),
):
    from app.services.supabase_client import get_saved_course_by_id, save_study_plan
    from app.services.gemini import generate_study_plan as gemini_plan

    course = await get_saved_course_by_id(body.course_id)
    parsed_syllabus = course.get("syllabus_parsed")
    plan = await gemini_plan(parsed_syllabus, body.weeks_available)
    await save_study_plan(body.course_id, plan)
    return StudyPlanResponse(course_id=body.course_id, plan_content=plan)


@router.get("/{course_id}", response_model=StudyPlanResponse)
async def get_study_plan(course_id: str):
    from app.services.supabase_client import get_study_plan
    plan = await get_study_plan(course_id)
    return StudyPlanResponse(course_id=course_id, plan_content=plan)
